src/axolotl/integrations/kernels/libs/sonicmoe/sf_layout.py

```
@@ -0,0 +1,158 @@
+def pack_scales_blocked(scale_2d: torch.Tensor) -> torch.Tensor:
+    """Rearrange ``(mn, sf_k)`` or ``(l, mn, sf_k)`` scales into ``(l, rm, rk, 512)``.
+
+    Works for any 1-byte scale dtype (e4m3 for NVFP4, e8m0 for MX). Pads mn to
+    a multiple of 128 and sf_k to a multiple of 4 with zeros.
+    """
+    if scale_2d.dim() == 2:
+        scale_2d = scale_2d.unsqueeze(0)
+    assert scale_2d.dim() == 3, f"expected (l, mn, sf_k), got {tuple(scale_2d.shape)}"
+    assert scale_2d.element_size() == 1, "scale dtype must be 1 byte (e4m3/e8m0/u8)"
+    orig_dtype = scale_2d.dtype
+    l, mn, sf_k = scale_2d.shape
```

Ruff (0.15.20)

[error] Ambiguous variable name: `l`

(E741)